### PR TITLE
Kill after 10 seconds of being near the fountain

### PR DESCRIPTION
**Pull request by Niker323**
https://github.com/dota2unofficial/12v12/issues/47

### CODE REVIEW

**Comment by fcalife** [comment]
This is not a very good band-aid, I would suggest killing any units that go out of bounds for anything over 1sec. Use the map's dimensions and GetAbsOrigin() to check for being out of bounds.

File: game/scripts/vscripts/addon_game_mode.lua [hunk 1]
```diff
@@ -71,6 +71,18 @@ function CMegaDotaGameMode:InitGameMode()
...
 		end
+		if event.abilityname == "item_blink" then
+			local oldpos = hero:GetAbsOrigin()
+			Timers:CreateTimer( 0.01, function()
+				local pos = hero:GetAbsOrigin()
+				local sum = pos.x + pos.y
+
+				if sum > 14150 or sum < -14350 or pos.x > 7750 or pos.x < -7750 or pos.y > 7500 or pos.y < -7300 then
+					FindClearSpaceForUnit(hero, oldpos, false)	
+					--CustomGameEventManager:Send_ServerToPlayer( PlayerResource:GetPlayer( event.PlayerID ), "display_custom_error", { message = "#" } )
```
Comment (ark120202) [inline game/scripts/vscripts/addon_game_mode.lua]:
```suggestion
```

File: game/scripts/vscripts/addon_game_mode.lua [hunk 2]
```diff
@@ -306,6 +318,24 @@ function CMegaDotaGameMode:OnThink()
 		self.m_CurrentXpScaleFactor = XP_SCALE_FACTOR_INITIAL + (xpFracTime * ( XP_SCALE_FACTOR_FINAL - XP_SCALE_FACTOR_INITIAL ) )
 --		print( "Gold scale = " .. self.m_CurrentGoldScaleFactor )
 --		print( "XP scale = " .. self.m_CurrentXpScaleFactor )
+
+		for i = 0, 23 do
+			if PlayerResource:IsValidPlayer( i ) then
+				local networth = 0
```
Comment (ark120202) [inline game/scripts/vscripts/addon_game_mode.lua]:
```suggestion
```

File: game/scripts/vscripts/addon_game_mode.lua [hunk 3]
```diff
@@ -306,6 +318,24 @@ function CMegaDotaGameMode:OnThink()
...
+				local hero = PlayerResource:GetSelectedHeroEntity( i )
+				if hero then
```
Comment (ark120202) [inline game/scripts/vscripts/addon_game_mode.lua]:
```lua
if hero and hero:IsAlive() then
```

**Comment by darklordabc** [comment]
Can we merge this now @ark120202?

File: game/scripts/vscripts/addon_game_mode.lua [hunk 4]
```diff
@@ -71,6 +71,17 @@ function CMegaDotaGameMode:InitGameMode()
 				duration = ability:GetSpecialValueFor("duration")
 			})
 		end
+		if event.abilityname == "item_blink" then
+			local oldpos = hero:GetAbsOrigin()
+			Timers:CreateTimer( 0.01, function()
+				local pos = hero:GetAbsOrigin()
+				local sum = pos.x + pos.y
```
Comment (ark120202) [inline game/scripts/vscripts/addon_game_mode.lua]: That can be extracted to something like `IsFountainBlockedPosition(pos)`

**Pull request merged**